Add lb route extension resource

maxi-cit opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->
Adds the new google_network_services_lb_route_extension resource.

Part of: https://github.com/hashicorp/terraform-provider-google/issues/17491
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/17372
<!--
Please self-review your PR against the review checklist before creating it: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

Completing the checklist will help speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.

If your PR is still work in progress, please create it in draft mode
-->

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/ for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_services_lb_route_extension`
```

```release-note:enhancement
networkservices: made required fields `extension_chains.extensions.authority ` and `extension_chains.extensions.timeout` optional in resource `google_network_services_lb_traffic_extension`
```

```release-note:enhancement
networkservices: made field `load_balancing_scheme` immutable in resource `google_network_services_lb_traffic_extension`, as in-place updating is always failing
```

```release-note:enhancement
networkservices: removed unsupported load balancing scheme `LOAD_BALANCING_SCHEME_UNSPECIFIED` from the field `load_balancing_scheme` in resource `google_network_services_lb_traffic_extension`
```

zli82016 commented:
Can you please fix the yaml lint errors? Thanks.

maxi-cit commented:
working on it

maxi-cit commented:
Hey @zli82016, I am still waiting for the API to become GA in order to validate the tests

PD: they told they are releasing it in the comming weeks

zli82016 commented:
> Hey @zli82016, I am still waiting for the API to become GA in order to validate the tests
> 
> PD: they told they are releasing it in the comming weeks

Thanks for letting me know.

zli82016 commented:
/gcbrun

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10119-old..auto-pr-10119) ( 1 file changed, 429 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10119-old..auto-pr-10119) ( 5 files changed, 1766 insertions(+), 2 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10119-old..auto-pr-10119) ( 1 file changed, 267 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_network_services_lb_route_extension` (1 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_network_services_lb_route_extension" "primary" {
  description = # value needed
  extension_chains {
    extensions {
      fail_open        = # value needed
      forward_headers  = # value needed
      supported_events = # value needed
    }
  }
  labels = # value needed
}

```

modular-magician commented:
#### Tests analytics
Total tests: `47`
Passed tests: `46`
Skipped tests: `0`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networkservices</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/dc214e5f-c245-4bcb-aab5-d99594da4fd0/build-log/recording_build/TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/dc214e5f-c245-4bcb-aab5-d99594da4fd0/recording/TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample.log)]


$\textcolor{red}{\textsf{Please fix these to complete your PR.}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/dc214e5f-c245-4bcb-aab5-d99594da4fd0/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/dc214e5f-c245-4bcb-aab5-d99594da4fd0/recording) for each test

zli82016 commented on mmv1/products/networkservices/LbRouteExtension.yaml:
```diff
@@ -0,0 +1,187 @@
+                  List of the HTTP headers to forward to the extension (from the client or backend).
+                  If omitted, all headers are sent. Each element is a string indicating the header name.
+                item_type: Api::Type::String
+              - !ruby/object:Api::Type::Array
```
Missing 'item_type'

maxi-cit replied:
added

zli82016 commented:
Can you also add the update tests for the fields that can be updated in-place? This is the doc [how to add an update test](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test). Thanks.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10119-old..auto-pr-10119) ( 1 file changed, 544 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10119-old..auto-pr-10119) ( 5 files changed, 2010 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10119-old..auto-pr-10119) ( 1 file changed, 267 insertions(+))

## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_network_services_lb_route_extension` (1 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_network_services_lb_route_extension" "primary" {
  description = # value needed
  extension_chains {
    extensions {
      forward_headers  = # value needed
      supported_events = # value needed
    }
  }
  labels  = # value needed
  project = # value needed
}

```

modular-magician commented:
#### Tests analytics
Total tests: `47`
Passed tests: `46`
Skipped tests: `0`
Affected tests: `1`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networkservices</li></ul></blockquote></details>

#### Action taken
<details> <summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/771945a2-db09-4ae5-9fc3-61ba35286446/recording/TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample.log)]


$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---
$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/771945a2-db09-4ae5-9fc3-61ba35286446/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/771945a2-db09-4ae5-9fc3-61ba35286446/recording) for each test

zli82016 commented on mmv1/products/networkservices/LbRouteExtension.yaml:
```diff
@@ -0,0 +1,187 @@
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'network_services_lb_route_extension_basic'
+    primary_resource_id: 'google_network_services_lb_route_extension'
```
The primary_resource_id can be simpler, like `default` or `lb_route_extension`

maxi-cit replied:
updated

zli82016 commented on mmv1/products/networkservices/LbRouteExtension.yaml:
```diff
@@ -0,0 +1,187 @@
+                description: |
+                  Specifies the timeout for each individual message on the stream. The timeout must be between 10-1000 milliseconds.
+                  A duration in seconds with up to nine fractional digits, ending with 's'. Example: "3.5s".
+                required: true
```
Is this field required? The API doc shows it is optional.

maxi-cit replied:
updated

zli82016 commented:
> ## Missing test report
> Your PR includes resource fields which are not covered by any test.
> 
> Resource: `google_network_services_lb_route_extension` (1 total tests) Please add an acceptance test which includes these fields. The test should include the following:
> 
> ```terraform
> resource "google_network_services_lb_route_extension" "primary" {
>   description = # value needed
>   extension_chains {
>     extensions {
>       forward_headers  = # value needed
>       supported_events = # value needed
>     }
>   }
>   labels  = # value needed
>   project = # value needed
> }
> ```


Can you add the tests for these fields?

Also, can you please add update tests for the fields that can be updated in-place? [Here](https://googlecloudplatform.github.io/magic-modules/develop/test/test/#add-an-update-test) is the doc. Thanks.

maxi-cit commented:
FYI
```
  | Error: Error creating LbRouteExtension: googleapi: Error 400: The request was invalid: usage of the supported_events field is not allowed for LbRouteExtension resource.
  | Details:
  | [
  |   {
  |     "@type": "type.googleapis.com/google.rpc.BadRequest",
  |     "fieldViolations": [
  |       {
  |         "field": "lb_route_extension.extension_chains[0].extensions[0].supported_events"
  |       }
  |     ]
  |   },
  |   {
  |     "@type": "type.googleapis.com/google.rpc.RequestInfo",
  |     "requestId": "834fddf60d27ec3a"
  |   }
  | ]
```

zli82016 commented:
> lb_route_extension

Got it. `supported_events` can be added later in Terraform provider when it will be supported in API side.

maxi-cit commented on mmv1/products/networkservices/LbRouteExtension.yaml:
```diff
@@ -0,0 +1,187 @@
+            The name for this extension chain. The name is logged as part of the HTTP request logs.
+            The name must conform with RFC-1034, is restricted to lower-cased letters, numbers and hyphens,
+            and can have a maximum length of 63 characters. Additionally, the first character must be a letter
+            and the last a letter or a number.
```
updated

maxi-cit commented on mmv1/products/networkservices/LbRouteExtension.yaml:
```diff
@@ -0,0 +1,187 @@
+                name: 'authority'
+                description: |
+                  The :authority header in the gRPC request sent from Envoy to the extension service.
+                required: true
```
you are right this is optional according to docs

maxi-cit commented:
Hello @zli82016, I added your suggestion as well with an update test. I also removed the required flag on the traffic extension resource  which was recently added for the same fields.

zli82016 commented:
/gcbrun

maxi-cit commented:
Hello @zli82016, are there any issue with the integration tests?

zli82016 commented:
> Hello @zli82016, are there any issue with the integration tests?

The integration tests are not triggered. I will check with out team.

zli82016 commented:
Can you please remove log.txt file? It could be the reason that the checked are not triggered.

maxi-cit commented:
Hello @zli82016, I guess that was the issue. I used that file to track progress on local tests. Sorry for pushing it.

zli82016 commented:
> Hello @zli82016, I guess that was the issue. I used that file to track progress on local tests. Sorry for pushing it.

No worries. The integration tests will be triggered soon.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10119-old..auto-pr-10119) ( 8 files changed, 2880 insertions(+), 15 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10119-old..auto-pr-10119) ( 8 files changed, 2880 insertions(+), 15 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10119-old..auto-pr-10119) ( 1 file changed, 256 insertions(+))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10119-old..auto-pr-10119) ( 4 files changed, 449 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `51`
Passed tests: `49`
Skipped tests: `0`
Affected tests: `2`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networkservices</li></ul></blockquote></details>


#### Action taken
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample|TestAccNetworkServicesLbRouteExtension_update </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/be86e29d-aee4-41d4-bcee-a5ac5d0a70d2/recording/TestAccNetworkServicesLbRouteExtension_networkServicesLbRouteExtensionBasicExample.log)]
`TestAccNetworkServicesLbRouteExtension_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/be86e29d-aee4-41d4-bcee-a5ac5d0a70d2/recording/TestAccNetworkServicesLbRouteExtension_update.log)]


$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---
$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/be86e29d-aee4-41d4-bcee-a5ac5d0a70d2/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/be86e29d-aee4-41d4-bcee-a5ac5d0a70d2/recording) for each test

zli82016 commented on mmv1/products/networkservices/LbRouteExtension.yaml:
```diff
@@ -0,0 +1,177 @@
+      All backend services and forwarding rules referenced by this extension must share the same load balancing scheme.
+        For more information, refer to Choosing a load balancer.
+    values:
+      - :LOAD_BALANCING_SCHEME_UNSPECIFIED
```
This value is not meaningless and can be removed.

maxi-cit replied:
removed

zli82016 commented on mmv1/products/networkservices/LbRouteExtension.yaml:
```diff
@@ -0,0 +1,177 @@
+    name: 'loadBalancingScheme'
+    description: |
+      All backend services and forwarding rules referenced by this extension must share the same load balancing scheme.
+        For more information, refer to Choosing a load balancer.
```
Can you please add a link to `Choosing a load balancer`?

Example:
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/accesscontextmanager/ServicePerimeter.yaml#L248

maxi-cit replied:
updated

zli82016 commented on mmv1/third_party/terraform/services/networkservices/resource_network_services_lb_route_extension_test.go:
```diff
@@ -0,0 +1,847 @@
+  description           = "my route extension"
+  location              = "us-west1"
+  load_balancing_scheme = "INTERNAL_MANAGED"
+  forwarding_rules      = [google_compute_forwarding_rule.default.self_link]
```
Can you please update the fields `forwarding_rules` with a list of values to test if the more items in the list are working?

maxi-cit replied:
I added a forwarding rule pointing to same proxy in order to lighten this test

zli82016 commented on mmv1/third_party/terraform/services/networkservices/resource_network_services_lb_route_extension_test.go:
```diff
@@ -0,0 +1,847 @@
+  name                  = "tf-test-l7-ilb-route-ext%{random_suffix}"
+  description           = "my route extension"
+  location              = "us-west1"
+  load_balancing_scheme = "INTERNAL_MANAGED"
```
Can you please update the fields `load_balancing_scheme` with a different value?

maxi-cit replied:
This value is not updateable. I am chaging it to immutable to allow update through recreation

zli82016 commented on mmv1/third_party/terraform/services/networkservices/resource_network_services_lb_route_extension_test.go:
```diff
@@ -0,0 +1,847 @@
+  }
+
+  labels = {
+    foo = "bar"
```
Can you please update the fields `labels` with a different value?

maxi-cit replied:
updated

zli82016 reviewed (changes requested):
LGTM mostly. Minor changes are needed.

maxi-cit commented:
Hello @zli82016, I added your suggestions and I think I am ready for another round of reviews. Could you run the tests please?

zli82016 commented:
> Hello @zli82016, any issues with the tests?

I just approved the test trigger. Let's see how it is going.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10119-old..auto-pr-10119) ( 8 files changed, 2910 insertions(+), 38 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10119-old..auto-pr-10119) ( 8 files changed, 2910 insertions(+), 38 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10119-old..auto-pr-10119) ( 1 file changed, 256 insertions(+))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10119-old..auto-pr-10119) ( 4 files changed, 449 insertions(+))

## Breaking Change(s) Detected

The following breaking change(s) were detected within your pull request.
- Field `load_balancing_scheme` changed from optional to required on `google_network_services_lb_traffic_extension` - [reference](https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes#field-optional-to-required)

If you believe this detection to be incorrect please raise the concern with your reviewer.
If you intend to make this change you will need to wait for a [major release](https://www.terraform.io/plugin/sdkv2/best-practices/versioning#example-major-number-increments) window.
An `override-breaking-change` label can be added to allow merging.

modular-magician commented:
#### Tests analytics
Total tests: `51`
Passed tests: `49`
Skipped tests: `0`
Affected tests: `2`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networkservices</li></ul></blockquote></details>


#### Action taken
<details> <summary>Found 2 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkServicesLbRouteExtension_update|TestAccNetworkServicesLbTrafficExtension_update </blockquote></details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccNetworkServicesLbRouteExtension_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/88e16eed-4d9f-4c8b-bf00-8f8d9eaf2a09/recording/TestAccNetworkServicesLbRouteExtension_update.log)]
`TestAccNetworkServicesLbTrafficExtension_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/88e16eed-4d9f-4c8b-bf00-8f8d9eaf2a09/recording/TestAccNetworkServicesLbTrafficExtension_update.log)]


$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---
$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/88e16eed-4d9f-4c8b-bf00-8f8d9eaf2a09/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/88e16eed-4d9f-4c8b-bf00-8f8d9eaf2a09/recording) for each test

zli82016 commented on mmv1/products/networkservices/LbTrafficExtension.yaml:
```diff
@@ -176,10 +174,12 @@ properties:
                 item_type: Api::Type::String
   - !ruby/object:Api::Type::Enum
     name: 'loadBalancingScheme'
+    required: true
```
It is a breaking change to make a field required.

Can you please create a new Github issue and then change it in Terraform Provider 6.0 major release?

maxi-cit replied:
Hello Zenhua, I know this is possible to be a breaking change but traffic extensions resource was merged very recently and I just wanted to reflect the same info as the [API doc](https://cloud.google.com/service-extensions/docs/reference/rest/v1beta1/projects.locations.lbTrafficExtensions#LbTrafficExtension:~:text=chains%20per%20resource.-,loadBalancingScheme,-enum%20(LoadBalancingScheme). So I dont think we need to wait to a major release to get this merged.

In case, we cannot omit this I think I better removed the required flag

zli82016 replied:
The traffic extensions resource is released to the user today and this PR will be released next week. If we make the breaking change, some users will see it. 

We still need to make the change in Provider 6.

maxi-cit replied:
Hello @zli82016, I updated the PR removing the flag and I will be opening an issue for this breaking change. do you kno the release time for Provider 6?

zli82016 replied:
Thanks. The service team can make the contribution from the end of June and the the release time is the end of August this year.

zli82016 commented:
I updated the release notes to reflect all of the changes in this PR.

zli82016 reviewed (changes requested):
Breaking change should be removed from this PR.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-10119-old..auto-pr-10119) ( 8 files changed, 2901 insertions(+), 29 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10119-old..auto-pr-10119) ( 8 files changed, 2901 insertions(+), 29 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-10119-old..auto-pr-10119) ( 1 file changed, 256 insertions(+))
Open in Cloud Shell: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-10119-old..auto-pr-10119) ( 4 files changed, 449 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: `51`
Passed tests: `51`
Skipped tests: `0`
Affected tests: `0`

<details><summary>Click here to see the affected service packages</summary><blockquote><ul><li>networkservices</li></ul></blockquote></details>

$\textcolor{green}{\textsf{All tests passed!}}$
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10119/artifacts/447c5fea-d224-45de-b6e3-603a011b3cdb/build-log/replaying_test.log)

zli82016 reviewed (approved):
LGTM. Thanks.
